src/components/workspaces/WorkspacesTable.tsx

```diff
@@ -78,14 +74,14 @@ const WorkspacesTable: React.FC<WorkspacesTableProps> = ({
     }
   }
 
-  const resource = currentUserOnly
-    ? createUserWorkspacesResource(sub)
-    : workspacesResource
+  const dataFilter = (data: any[]) =>
+    data.filter((workspace) => (workspace as Workspace).author.id === sub)
 
   return (
     <DataTableSuspense
       columns={columns}
-      resource={resource}
+      resource={workspacesResource}
+      filter={currentUserOnly ? dataFilter : undefined}
```

Comment:
I typically tend to use rather vanilla empty implementations than undefined-checks on both client and server side, but it is probably more a matter of taste.